src/arcaflow_plugin_sdk/schema.py

```diff
@@ -5989,10 +5989,16 @@ def __call__(
             input.validate(params, tuple(["input"]))
         # Run the step
         result = self._handler(step_local_data.initialized_object, params)
+        if not isinstance(result, tuple):
+            raise BadArgumentException(
+                f"The implementation of step {run_id}/{self.id} returned"
+                f" type {type(result)}; expected a tuple with two"
```

Comment:
You added `result` to the "tuple but not twofer" path, and I have no idea what we'd see here if it's not even a tuple ... but the "expected tuple with two values [etc]" still feels like an unfair and annoying tease. I guess at least you're reporting the actual type, and we could end up with pages of output on a `{result!r}` ... so maybe this is the best we can reasonable expect. 😦

Reply:
Does the latest change address your concern?